### DISCUSSION TITLE
Improving Error Message for missing data

### DISCUSSION THREAD

**Discussion by jinwar** [DASDAE/dascore#65] [Ideas]
Current error messages for fetching a time section without data are not very clear.

**Comment by d-chambers** [comment]
Are you referring to when you try to index into a spool that has already been selected, or is the error raised on the `Spool.select` method call? 

```python
import dascore as dc

spool = dc.get_example_spool()

# does the unclear error message pop up here? 
no_data_spool = spool.select(time=('2030-01-01", None))

# or here?
patch = no_data_spool[0]
```

**Comment by d-chambers** [comment]
I figured out why this is happening.

For some Terra15 data the gitter in the GPS measurements cause subsequent samples to be reported at a early time than the previous sample.

Here is an example: 

![image](https://user-images.githubusercontent.com/11671536/200059543-6f1a5b0f-9b09-4185-ac54-274eb54b8427.png)

The y axis is the time difference between sample n and n +1, the x axis is just sample number.

When this happens, the index is no longer treated as a monotonic range but a collection of values and goes screwy. I will look into the terra15 parser and think about how to fix it.

**Comment by d-chambers** [comment]
Here is a snippet to reproduce the `KeyError`. The main factor is a non-monotonic index.

```python
import numpy as np
import pandas as pd

index = np.random.rand(100) * 1_000_000 * np.timedelta64(1, 'ns') + np.datetime64('2020-01-10')

dt_index = pd.DatetimeIndex(index, name='time') 

time_1 = np.datetime64('2020-01-10')
time_2 = time_1 + np.timedelta64(1, 's')

dt_index.slice_locs(time_1, time_2)
```
